tree_unique_args/src/lib.rs

```diff
@@ -35,6 +36,11 @@ pub enum Expr {
     Or(Box<Expr>, Box<Expr>),
     Not(Box<Expr>),
     Get(Box<Expr>, usize),
+    /// Concat is a convenient built-in way
+    /// to put two tuples together.
+    /// It's not strictly necessary, but
+    /// doing it by constructing a new big tuple is tedius and slow.
+    Concat(Box<Expr>, Box<Expr>),
```

Comment:
Why is this an `Expr` instead of a `fn(Expr, Expr) -> Expr`?

Reply:
This PR proposes adding a concat operator to our encoding.
It can't be a function on two exprs, since the two exprs haven't been evaluated yet. For example: `(Concat (Arg 0) (Arg 0))`